### ISSUE TITLE
DiceCoefficient and DiceLoss 

### ISSUE THREAD

**Issue by Littlehhh** [wolny/pytorch-3dunet#38]
## About DiceCoefficient and DiceLoss 
Thanks for sharing your work. I have read and learn your source code, It's helpful.    
But there is a problem about DiceCoefficient and DiceLoss .  

In  "unet3d/metrics.py"`class DiceCoefficient` the commit says that `DO NOT USE this metric when training with DiceLoss, otherwise the results will be biased towards the loss.`   Is it  because the `class DiceLoss` and `class DiceCoefficient` use the same `function compute_per_channel_dice(input, target, epsilon=1e-5, ignore_index=None, weight=None)` in "unet3d/losses.py"?

**Comment by wolny** [comment]
Hi @Littlehhh,
I'm glad that you found the code to be useful. Regarding your question: the reason not to use DiceCoefficient as an evaluation metric when the model is trained with the DiceLoss is simply because it's unfair: the two functions are related (i.e. DiceLoss = 1 - DiceCoefficient). In order to compare fairly the evaluation function should be depend on the task at hand and be independent of the loss function. E.g. imagine that you're training 2 networks on your data: 1. using the DiceLoss 2. using the CrossEntropy loss. Then if you use DiceCoefficient to evaluate the models performance on a held out validation set, then model 1 one will always be better than model 2, since model 1 was pretty much trained to optimized the validation metric itself. It might be that model 2 actually performs and generalizes better, but you're not able to spot that because the evaluation metric is 'exploited' by model 1. Hope that makes it clear.

Best
